flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -120,15 +120,20 @@
 	/** True iff the component has been started. */
 	private boolean started;
 
+	/** Release task executor only when each produced result partition is either consumed or failed.. */
+	private final boolean waitResultConsumedToRelease;
```

Comment:
nit: rename everywhere `waitResultConsumedToRelease` to `waitResultConsumedBeforeRelease`?